### PR TITLE
fix issue in find element error handling, so implicit wait can work

### PR DESCRIPTION
**Pull request by jlipps**
cc @mykola-mokhnach: unfortunately we can't do `instanceof` checks where the LHS could be an instance of a class from a different version of a module as the RHS.

### CODE REVIEW

File: lib/commands/find.js [hunk 1]
```diff
@@ -35,7 +35,17 @@ helpers.findElOrEls = async function (strategy, selector, mult, context = '') {
...
+      // if the error that comes back is from a proxied request, we need to
+      // unwrap it to its actual protocol error first
+      if (_.isFunction(err.getActualError)) {
+        err = err.getActualError(); // eslint-disable-line no-ex-assign
+      }
+
+      // now we have to inspect the error to determine if it is a no such
+      // element error, based on the shape of the error object from
+      // appium-base-driver
+      if (err.jsonwpCode === 7 || err.error === "no such element") {
```
Comment (mykola-mokhnach) [inline lib/commands/find.js]: why should we use magic numbers here? I'd rather use `errors.NoSuchElementError.error()/code()`

**Comment by mykola-mokhnach** [comment]
@jlipps Thanks for pointing that out. I didn't know

File: lib/commands/find.js (hunk 1 above)
Comment (jlipps) [reply lib/commands/find.js]: right you are! will fix

**Comment by jlipps** [comment]
@mykola-mokhnach yeah i learned this the hard way from the early days of writing basedriver. can you imagine the crazy bugs that result from this? it's why we have [isErrorType()](https://github.com/appium/appium-base-driver/blob/master/lib/protocol/errors.js#L747)... actually come to think of it, we could just export this function and use it

**Comment by mykola-mokhnach** [comment]
yep, I was actually wondering why such function exists. It might be useful to add a docstring there with a detailed purpose explanation, otherwise this limitation will stay in your head only ;)

Also, you say it is safe to use `instanceof` in scope of a single package, so https://github.com/appium/appium-base-driver/blob/165543e16dc8ad2748d9ce0316b5ec0b5856eaae/lib/jsonwp-proxy/proxy.js#L164 should work?

**Comment by jlipps** [comment]
@mykola-mokhnach that will only be safe it is guaranteed that `err` is only ever instantiated from within basedriver. In other words, at some point there is a `new ProxyRequestError(...)` which results in the error which is checked here. And that instantiation must also be in basedriver. Can we guarantee this?

**Comment by jlipps** [comment]
published in 3.0.3

**Pull request merged**